pkg/k8s/wrapper.go

```diff
@@ -220,3 +224,59 @@ func (k *k8sWrapper) ListEvents(ops []client.ListOption) (*eventsv1.EventList, e
 	}
 	return events, nil
 }
+
+func (k *k8sWrapper) GetCNINode(cniNodeName, cniNodeNamespace string) (*rcv1alpha1.CNINode, error) {
+	cninode := &rcv1alpha1.CNINode{}
+	if err := k.cacheClient.Get(k.context, types.NamespacedName{
+		Name:      cniNodeName,
+		Namespace: cniNodeNamespace,
+	}, cninode); err != nil {
+		return cninode, err
+	}
+	return cninode, nil
+}
+
+func (k *k8sWrapper) CreateCNINode(node *v1.Node) error {
+	err := retry.OnError(retry.DefaultBackoff,
```

Comment:
Why do we implement retry inside of this client, instead of relying on the standard reconciler retry mechanisms?

Reply:
I think letting reconciler to retry makes more sense in this case. I will do ignore exists and let reconciler retry on error.